view.php

```diff
@@ -20,32 +20,46 @@
 require_once($CFG->libdir.'/tablelib.php');
 
 $id = required_param('id', PARAM_INT);
+$download = optional_param('download', false, PARAM_ALPHA);
```

Comment:
This should be PARAM_BOOL

Reply:
The table download doesn't work if it's PARAM_BOOL. When you download the table, the 'download' param is used to identify the requested format.

Reply:
Ah gottcha, the problem with RAW is that it doesn't get any validation at all so you need to be very careful with what you do with it. So PARAM_TEXT is better in this case.